src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php

```diff
@@ -65,11 +70,33 @@ public function onKernelResponse(FilterResponseEvent $event)
         $response = $event->getResponse();
         $request = $event->getRequest();
 
-        if ($response->headers->has('X-Debug-Token') && null !== $this->urlGenerator) {
+        $hasProfile = $this->profileStack instanceof ProfileStack ? $this->profileStack->has($request) : $response->headers->has('X-Debug-Token');
+
+        if ($hasProfile && null !== $this->urlGenerator) {
+            if ($this->profileStack instanceof ProfileStack) {
+                $profile = $this->profileStack->get($request);
+
+                $token = $profile->getToken();
+
+                foreach ($profile->getCollectors() as $collector) {
+                    if ($collector instanceof ExceptionDataCollector && $collector->hasException()) {
+                        $panel = $collector->getName();
+
+                        break;
+                    } elseif ($collector instanceof DumpDataCollector && $collector->getDumpsCount() > 0) {
+                        $panel = $collector->getName();
+                    }
+                }
+            } else {
+                $token = $response->headers->get('X-Debug-Token');
+            }
             try {
                 $response->headers->set(
                     'X-Debug-Token-Link',
-                    $this->urlGenerator->generate('_profiler', ['token' => $response->headers->get('X-Debug-Token')], UrlGeneratorInterface::ABSOLUTE_URL)
+                    $this->urlGenerator->generate('_profiler', [
+                        'token' => $token,
+                        'panel' => $panel ?? null,
```

Comment:
you should define `$panel` as `null` at the beginning of the logic instead of dealing with an undefined variable here.